tss-esapi/src/context/tpm_commands/object_commands.rs

```diff
@@ -71,7 +71,7 @@ impl Context {
                 self.optional_session_2(),
                 self.optional_session_3(),
                 &sensitive_create,
-                &public.clone().into(),
+                &public.clone().try_into()?,
```

Comment:
This is not directly related to your PR but the `clone()` call... ugh... maybe we should track or decide on whether to take the Public by value (so that the caller has to explicitly clone if they want to reuse public). Because currently it always clones public even if passing by value a single thing would suffice...

Reply:
So, I didn't notice this before in the Rust API guidelines, but it seems to actually be specified there: [we should ask for ownership instead of cloning from a reference](https://rust-lang.github.io/api-guidelines/flexibility.html#caller-decides-where-to-copy-and-place-data-c-caller-control). Given that this essentially breaks the API, the question is whether we want to ship it in 7.0.0. Thoughts? Should we discuss it under #290 ? cc @Superhepper

Also, I'm not sure if I actually answered the question or if you meant that we could not take ownership at all, since we only need a reference in the end (though the reference is to `TPM2B_PUBLIC`)

Reply:
Damn, deleted my comment by accident...

It seems, from the Rust API guidelines, that [we should ask for ownership of the value instead of cloning from a reference](https://rust-lang.github.io/api-guidelines/flexibility.html#caller-decides-where-to-copy-and-place-data-c-caller-control). Since this breaks the API we need to decide whether to do this for 7.0.0 or leave it for a subsequent release. It's annoying, but it's not the end of the world, and we have a lot of changes due in 7.0.0, so I'd prefer sticking with the current API, but can be convinced otherwise. Thoughts? cc @Superhepper 

Also, I don't know if the paragraph above answers your question - were you asking if we really need ownership here at all, since we only need to pass down a reference?

Reply:
There are two things in play here:
  - first, conversion from `&Public` to `Public` that allows to omit the `clone()` call, that was what I was talking about, I don't think this is a big deal with regards to API breakage (I've been updating my crate continuously to 7-alpha and there were bigger issues but I understand if you'd like to defer that),
  - then I thought: it's actually just doing `try_into` to convert to `PublicBuffer` (checked docs for `TPM2B_PUBLIC`) so a next question raises: why not take `PublicBuffer` parameter instead?

Reply:
We have already declared that there will be some huge interface changes in 7.0.0. And I think I would prefer that we do all the interface changes that we have been wanted to do now when we have the chance. I am already prepared that moving my code to 7.0.0 will be a huge task. And it would annoy me if I now fixed all my code and then some one makes interfaces changes for lets say 8.0.0 and new functionality that I need are based on those changes.

I am not to found of the method API dealing with PublicBuffer. Because, and correct me if I am wrong, that is not a type that is returned from the context methods. Public is the type we uses because it contains the values in a human readable form, PublicBuffer is something you use if you want the type as bytes.

Reply:
> it's actually just doing try_into to convert to `PublicBuffer`

It's not, that's why I implemented `TryFrom<Public> for TPM2B_PUBLIC`, so we don't have to go through `PublicBuffer`, because getting a `PublicBuffer` (or getting something from `PublicBuffer`) _always_ requires either marshalling or unmarshalling, so you'd have two useless operations in-between.

Reply:
Oh, great! I must've overlooked that fact. Thanks for the explanation!